ckan/config/middleware/__init__.py

```diff
@@ -96,10 +100,58 @@ def ask_around(self, environ):
 
         return answers
 
+    def handle_i18n(self, environ):
+        '''
+        Note: This function used to be the I18nMiddleware.
+
+        Strips the locale code from the requested url
+        (eg '/sk/about' -> '/about') and sets environ variables for the
+        language selected:
+
+            * CKAN_LANG is the language code eg en, fr
+            * CKAN_LANG_IS_DEFAULT is set to True or False
+            * CKAN_CURRENT_URL is set to the current application url
+        '''
+
+        # We only update once for a request so we can keep
+        # the language and original url which helps with 404 pages etc
+        if 'CKAN_LANG' not in environ:
+            path_parts = environ['PATH_INFO'].split('/')
+            if len(path_parts) > 1 and path_parts[1] in self.locale_list:
```

Comment:
this ignores the `root_path` setting http://docs.ckan.org/en/latest/maintaining/configuration.html?highlight=root_path#ckan-root-path